addons/event-audit/common/src/main/java/org/commonjava/indy/event/audit/EventAuditListener.java

```diff
@@ -0,0 +1,227 @@
+import org.commonjava.propulsor.content.audit.model.FileGroupingEvent;
+import org.commonjava.propulsor.content.audit.model.FileGroupingEventType;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.enterprise.context.ApplicationScoped;
+import javax.enterprise.event.Observes;
+import javax.inject.Inject;
+import java.util.Date;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Set;
+
+@ApplicationScoped
+public class EventAuditListener
+{
+
+    @Inject
+    DownloadManager downloadManager;
+
+    @Inject
+    ContentDigester contentDigester;
+
+    @Inject
+    StoreDataManager storeManager;
+
+    @Inject
+    ISPNEventPublisher eventPublisher;
+
+    @Inject
+    PromoteChangeManager promoteChangeManager;
+
+    @Inject
+    EventAuditConfig eventAuditConfig;
+
+    private final Logger logger = LoggerFactory.getLogger( "org.commonjava.indy.event.audit" );
+
+    public void onFileAccess( @Observes final FileAccessEvent event )
+    {
+
+        if ( !eventAuditConfig.isEnabled() )
+        {
+            return;
+        }
+
+        FileEvent fileEvent = new FileEvent( FileEventType.ACCESS );
+        transformFileEvent( event, fileEvent );
+        eventPublisher.publishFileEvent( fileEvent );
+
+    }
+
+    public void onFileUpload( @Observes final FileStorageEvent event )
+    {
+
+        if ( !eventAuditConfig.isEnabled() )
+        {
+            return;
+        }
+
+        FileEvent fileEvent = new FileEvent( FileEventType.STORAGE );
+        transformFileEvent( event, fileEvent );
+        eventPublisher.publishFileEvent( fileEvent );
+    }
+
+    public void onPromoteComplete( @Observes final PromoteCompleteEvent event )
+    {
+        if ( !eventAuditConfig.isEnabled() )
+        {
+            return;
+        }
+
+        FileGroupingEvent fileGroupingEvent = new FileGroupingEvent( FileGroupingEventType.BY_PATH_PROMOTION );
+        transformFileGroupingEvent( event, fileGroupingEvent );
+        eventPublisher.publishFileGroupingEvent( fileGroupingEvent );
+    }
+
+    private void transformFileEvent( org.commonjava.maven.galley.event.FileEvent event, FileEvent fileEvent )
+    {
+        EventMetadata metadata = event.getEventMetadata();
+        final TrackingKey trackingKey = (TrackingKey) metadata.get( "tracking-id" );
+        if ( trackingKey == null )
+        {
+            logger.trace( "No tracking key. Skip audit." );
+            return;
+        }
+
+        Transfer transfer = event.getTransfer();
+        if ( transfer == null )
+        {
+            logger.trace( "No transfer. Skip audit." );
+            return;
+        }
+
+        fileEvent.setTargetPath( transfer.getPath() );
+        fileEvent.setNodeId( "" );
+        fileEvent.setSessionId( trackingKey.getId() );
+        fileEvent.setTimestamp( new Date() );
+
+        final KeyedLocation keyedLocation = (KeyedLocation) transfer.getLocation();
+        final String path = transfer.getPath();
```

Comment:
As this path is defined with transfer.getPath, suggest to use it all instead of using transfer.getPath() again in following code. (seems it is called once in previous code too)